test/gas-analytics/HubPool.RefundExecution.ts

```diff
@@ -0,0 +1,136 @@
+import {
+  toBNWei,
+  toBN,
+  SignerWithAddress,
+  seedWallet,
+  Contract,
+  ethers,
+  getContractFactory,
+  BigNumber,
+  randomAddress,
+  expect,
+} from "../utils";
+import * as consts from "../constants";
+import { TokenRolesEnum } from "@uma/common";
+import { hubPoolFixture, enableTokensForLP } from "../HubPool.Fixture";
+import { buildPoolRebalanceLeafTree, buildPoolRebalanceLeafs } from "../MerkleLib.utils";
+
+require("dotenv").config();
+
+let hubPool: Contract, timer: Contract, weth: Contract, mockAdapter: Contract, mockSpoke: Contract;
+let owner: SignerWithAddress, dataWorker: SignerWithAddress, liquidityProvider: SignerWithAddress;
+
+// Data structure to create pool rebalance leafs:
+const l1Tokens: { [key: number]: Contract[] } = {};
+const destinationChainIds: number[] = [];
+const _bundleLpFeeAmounts: BigNumber[][] = [];
+const _netSendAmounts: BigNumber[][] = [];
+
+// Constants caller can tune to modify gas tests
+const REFUND_TOKEN_COUNT = 10;
+const REFUND_CHAIN_COUNT = 10;
+const REFUND_SEND_COUNT = 10; // Send amount per destination chain + token combo
+const SEND_AMOUNT = toBNWei("10");
+const STARTING_LP_AMOUNT = SEND_AMOUNT; // This should be >= `SEND_AMOUNT` otherwise some relays will revert because
+// the pool balance won't be sufficient to cover the relay.
+const LP_FEE = SEND_AMOUNT.div(toBN(10));
+
+async function deployErc20(signer: SignerWithAddress, tokenName: string, tokenSymbol: string) {
+  const erc20 = await (await getContractFactory("ExpandedERC20", signer)).deploy(tokenName, tokenSymbol, 18);
+  await erc20.addMember(TokenRolesEnum.MINTER, owner.address);
+  return erc20;
+}
+
+// Construct tree with REFUND_CHAIN_COUNT leaves, each containing REFUND_TOKEN_COUNT sends
+async function constructSimpleTree() {
```

Comment:
Before adding more code to this script and more test cases, I'm curious what you guys think about this being on the right track for gas analysis? Currently it only allows you create `REFUND_CHAIN_COUNT` count of leaves, each with `REFUND_TOKEN_COUNT` rebalances triggered by the `HubPool.executeRelayerRefund` method